### DISCUSSION TITLE
Running Maven plugins which work only with Java 8 as part of the Maven Project Builder

### DISCUSSION THREAD

**Discussion by digulla** [eclipse-m2e/m2e-core#812] [Q&A]
I have an old project which uses jaxb2-maven-plugin. The verson we use only works with Java 8 (missing XML classes in 11 and up).

Since Eclipse 2022-06 (4.24) uses Java 17, I'm getting errors in the project since the plugin fails.

Is there a trick to get m2e to run the plugins with a different Java version than the IDE? Or do I have to upgrade jaxb2-maven-plugin? (I can't upgrade the whole project to 11, so I'd need to find a newer version of jaxb2 which runs with both versions).

**Comment by laeubi** [comment]
> Is there a trick to get m2e to run the plugins with a different Java version than the IDE?

Not really, you can run the m2e maven build with a different java but not the plugin itself, as m2e is running "inside" eclipse

> Or do I have to upgrade jaxb2-maven-plugin? (I can't upgrade the whole project to 11, so I'd need to find a newer version of jaxb2 which runs with both versions).

You should upgrade your **build** to run with java 11, the **project** can still be java 8.

**Comment by laeubi** [comment]
> missing XML classes in 11 and up

By the way it should be possible to add missing stuff to the `<dependency>` section of the plugin

**Reply by digulla** [reply]
Great idea. I'll try that.
